packages/compass-components/src/components/collapsible-field-set.tsx

```diff
@@ -45,10 +35,7 @@ const UnthemedCollapsibleFieldSet = ({
 }: React.PropsWithChildren<CollapsibleFieldSetProps>): React.ReactElement => {
   const labelId = dataTestId || 'collapsible-fieldset-props';
```

Comment:
The fact that this relies on test id to set correct, not duplicated `id` and `for` attribute is a bit concerning, you have two of these on the screen without test id and now you have two identical ids passed to the html elements. Please let's use and explicit required prop or `useId` hook here, it's specifically what it was made for